lib/Doctrine/ORM/Tools/SchemaTool.php

```diff
@@ -956,10 +977,18 @@ public function getUpdateSchemaSql(array $classes, $saveMode = false)
         $schemaDiff = $comparator->compareSchemas($fromSchema, $toSchema);
 
         if ($saveMode) {
+            if (! method_exists($schemaDiff, 'toSaveSql')) {
+                throw GlobalNotSupported::createForDbal4('Using save mode when getting update schema SQL');
+            }
```

Comment:
This looks like dead code to me. We don't support DBAL 4 on this branch and on the 3.0.x branch, we're going to remove the `$saveMode` parameter anyway.